MvvmCross/ViewModels/MvxNavigationViewModel.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MS-PL license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using MvvmCross.Logging;
+using MvvmCross.Navigation;
+
+namespace MvvmCross.ViewModels
+{
+    public abstract class MvxNavigationViewModel
+        : MvxViewModel
+    {
+        private IMvxLogProvider _logProvider;
+        private IMvxLog _log;
+
+        protected MvxNavigationViewModel(IMvxLogProvider logProvider, IMvxNavigationService navigationService) : base()
```

Comment:
Should be public?

Reply:
How would you be able to create a new instance in a unit test?

Reply:
You can mock it

Reply:
Done - I've added a basic unit test to validate that protected properties are indeed set during construction.